File: articles/jung-alchemy-and-symbolic-transformation/go/alchemical_transformation.go
package main

import (
	"fmt"
	"math"
)

func transformationScore(nigredoPressure, albedoClarity, rubedoVitality, vesselStrength, onesidedness, mercurialVolatility float64) float64 {
	return 0.42*nigredoPressure +
		0.55*albedoClarity +
		0.66*rubedoVitality +
		0.58*vesselStrength -
		0.60*onesidedness -
		0.30*math.Abs(mercurialVolatility)
}

func main() {
	score := transformationScore(0.70, 0.52, 0.44, 0.58, 0.62, 0.84)
	fmt.Printf("Synthetic alchemical transformation score: %.3f\n", score)
}

File: articles/jung-alchemy-and-symbolic-transformation/rust/alchemical_transformation.rs
fn transformation_score(
    nigredo_pressure: f64,
    albedo_clarity: f64,
    rubedo_vitality: f64,
    vessel_strength: f64,
    onesidedness: f64,
    mercurial_volatility: f64,
) -> f64 {
    0.42 * nigredo_pressure
        + 0.55 * albedo_clarity
        + 0.66 * rubedo_vitality
        + 0.58 * vessel_strength
        - 0.60 * onesidedness
        - 0.30 * mercurial_volatility.abs()
}

fn main() {
    let score = transformation_score(0.70, 0.52, 0.44, 0.58, 0.62, 0.84);
    println!("Synthetic alchemical transformation score: {:.3}", score);
}
